---
title: "Sunil Ex17 Phospho differential expression & KSEA"
author: "Camilo Posso"
date: "08/11/2021"
output: 
  html_document:
    code_folding: hide
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, include=FALSE}
library(amlresistancenetworks)
library(kableExtra)
library(dplyr)
library(readr)
library(readxl)
source("../../Util/synapseUtil.R")

# corrected for loading mass and plexID, long time and subtle
p.site.corrected <- querySynapseTable("syn26047149")

phos.dat <- p.site.corrected
phos.dat$Peptide <- sub("^.*@(.*)$", "\\1", phos.dat$Accession)
phos.dat$site <- sub("^(.*)@.*$", "\\1", phos.dat$Accession)
phos.dat$Gene <- sub("(.*)-.*$", "\\1", phos.dat$site)

## We have the metadata selected from the long form table
summary <- p.site.corrected[p.site.corrected$Accession == "AAK1-T606t@K.VQTT*PPPAVQGQK.V", 
                            c("Sample", "BeatAML Patient ID", 
                              "Plex", "Loading Mass", "Description", 
                              "Period", "Treatment")]

rownames(summary) <- summary$Sample

## Organizing time points into the right order (not alphanumeric)
time.points <- c("P2 Pre-Study", "P2 Cycle 1 Day 1", "P2 Cycle 1 Day 28", "P2 Cycle 2 Day 28", "P2 Cycle 3 Day 28", 
                 "P1B Cycle 1 Day 1", "P1B Cycle 1 Day 28", "P1B Cycle 2 Day 28", "P1B Cycle 3 Day 28", "P1B Cycle 6 Day 28",
                 "P1B Cycle 9 Day 28", "P1B Conc Cycle 9 Day 28") %>%
  factor(.)


summary$Description <- factor(summary$Description, levels = time.points)
summary <- summary[sort(summary$Sample), ]
summary$`BeatAML Patient ID` <- factor(summary$`BeatAML Patient ID`, levels = unique(summary$`BeatAML Patient ID`))
summary$Treatment <- make.names(summary$Treatment)
```

### Introduction

We access synapse to load the data in long form. We use data which has been corrected for both
Loading mass effect, as well as plex effect. Below we see a metadata table for this experiment.

```{r pressure, echo=FALSE}
show <- summary 
kbl(show) %>%
  kable_paper("hover", full_width = F)

```

In this report, we collect the results of rank based GSEA, as well as differential expression. We perform
5 comparisons overall. First, we compare the Healthy group to the Pre-Treatment group, the Gilteritinib group, and the 
Decitabine group. As some patients are over-represented in these groups (such as patient 5104), we choose the *earliest*
representative sample per patient per group.

```{r prepare groups, limma, and early vs late ksea, include=FALSE}
##select the samples of interest
healthy <- subset(summary, `BeatAML Patient ID` == 'Healthy Donor Stroma')

pre.treatment <- subset(summary, Period == "Pre-Treatment")

## Earliest sample, per patient, which has been treated with GILT
treated.gilt <- subset(summary, Treatment == "GILT") %>%
  filter(`BeatAML Patient ID` != "5029") %>%
  group_by(`BeatAML Patient ID`) %>%
  slice(1)

## Earliest sample, per patient, which has been treated with GILT + DEC
treated.gilt.and.dec <- subset(summary, Treatment == "GILT...DEC") %>%
  group_by(`BeatAML Patient ID`) %>%
  slice(1)

# treated <- subset(summary, Type == 'Treated')%>%
#   select(Sample) %>%
#   unique()
#
# treated.late <- subset(summary, Period == "Late")%>%
#   select(Sample)%>%
#   unique()
# 
# treated.early <- subset(summary, Period == "Early") %>%
#   group_by(`BeatAML Patient ID`) %>%
#   slice(1) %>%
#   ungroup(`BeatAML Patient ID`) %>%
#   select(Sample) %>%
#   unique()

##then we spread the proteomics data into a matrix
phos.mat <- phos.dat%>%
  select(LogRatio, Sample, site)%>%
  tidyr::pivot_wider(values_from='LogRatio',names_from='Sample',
                      values_fn=list(LogRatio=mean))%>%
  tibble::column_to_rownames('site')

gene.to.site<-dplyr::select(phos.dat,Gene,site,Peptide)%>%distinct()%>%
    dplyr::mutate(residue=stringr::str_replace(site,paste0(Gene,'-'),''))%>%
    dplyr::mutate(residue=stringr::str_replace_all(residue,"([STY])", ";\\1"))%>%
    dplyr::mutate(residue=stringr::str_replace(residue,"^;", ""))%>%
    dplyr::mutate(residue=stringr::str_replace_all(residue,"([sty])", ""))

#### Comparisons to Healthy cells ####

## LogFC computed as pre-treatment - healthy
phos.diffex.pre <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, pre.treatment$Sample)

## LogFC computed as treated GILT - healthy.
phos.diffex.gilt <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, treated.gilt$Sample)

## LogFC computed as treated GILT+DEC - healthy
phos.diffex.gilt.dec <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, treated.gilt.and.dec$Sample)


#### Paired comparisons for Pre vs GILT and GILT vs GILT + DEC ####
## Pre vs GILT. logFC computed as GILT - PRE
samples.compared <- rbind(pre.treatment, treated.gilt)
ID <- factor(samples.compared$`BeatAML Patient ID`, levels = unique(samples.compared$`BeatAML Patient ID`))
Treatment <- factor(samples.compared$Treatment, levels = c("PRE", "GILT"))
design <- model.matrix(~ID+Treatment)
fit <- lmFit(phos.mat[, samples.compared$Sample], design)
fit <- eBayes(fit)
phos.paired.gilt.vs.pre <- topTable(fit, coef = "TreatmentGILT", number = Inf) %>%
  mutate(featureID = rownames(.)) %>%
  arrange(P.Value)

### To make sure the moderated t-statistics resemble those from a t.test
# t.test(as.numeric(phos.mat["MAP1A-S2323sS2330s", treated.gilt$Sample]),
#        as.numeric(phos.mat["MAP1A-S2323sS2330s", pre.treatment$Sample]),
#        paired = T)

## GILT vs GILT+DEC. logFC computed as GILT...DEC - GILT
## Choosing only GILT samples with corresponding GILT+DEC 
treated.gilt.restricted <- subset(summary, Treatment == "GILT") %>%
  filter(`BeatAML Patient ID` %in% treated.gilt.and.dec$`BeatAML Patient ID`)
samples.compared <- rbind(treated.gilt.restricted, treated.gilt.and.dec)
ID <- factor(samples.compared$`BeatAML Patient ID`, levels = unique(samples.compared$`BeatAML Patient ID`))
Treatment <- factor(samples.compared$Treatment, levels = c("GILT", "GILT...DEC"))
design <- model.matrix(~ID+Treatment)
fit <- lmFit(phos.mat[, samples.compared$Sample], design)
fit <- eBayes(fit)
phos.paired.giltdec.vs.gilt <- topTable(fit, coef = "TreatmentGILT...DEC", number = Inf) %>%
  mutate(featureID = rownames(.)) %>%
  arrange(P.Value)

### To make sure the moderated t-statistics resemble those from a t.test
# t.test(as.numeric(phos.mat["MAP1A-S2323sS2330s", treated.gilt.and.dec$Sample]), 
#        as.numeric(phos.mat["MAP1A-S2323sS2330s", treated.gilt.restricted$Sample]),
#        paired = T)

# ### Equivalent way of testing Gilt vs Pre (paired t-test) using limma contrasts
# samples.compared <- rbind(pre.treatment, treated.gilt)
# ID <- factor(samples.compared$`BeatAML Patient ID`, levels = unique(samples.compared$`BeatAML Patient ID`))
# Treatment <- factor(samples.compared$Treatment, levels = c("PRE", "GILT"))
# design <- model.matrix(~0+Treatment+ID)
# fit <- lmFit(phos.mat[, samples.compared$Sample], design)
# contrast.matrix <- makeContrasts("TreatmentGILT-TreatmentPRE", levels = design)
# fit <- contrasts.fit(fit, contrast.matrix)
# fit <- eBayes(fit)
# res4 <- topTable(fit, coef = "TreatmentGILT-TreatmentPRE", number = Inf) %>%
#   mutate(featureID = rownames(.)) %>%
#   arrange(P.Value)
# 
# 
# ### Equivalent way of testing Gilt + Dec vs Gilt (paired t-test) using limma contrasts
# samples.compared <- rbind(treated.gilt.restricted, treated.gilt.and.dec)
# ID <- factor(samples.compared$`BeatAML Patient ID`, levels = unique(samples.compared$`BeatAML Patient ID`))
# Treatment <- factor(samples.compared$Treatment, levels = c("GILT", "GILT...DEC"))
# design <- model.matrix(~0+Treatment+ID)
# fit <- lmFit(phos.mat[, samples.compared$Sample], design)
# contrast.matrix <- makeContrasts("TreatmentGILT...DEC-TreatmentGILT", levels = design)
# fit <- contrasts.fit(fit, contrast.matrix)
# fit <- eBayes(fit)
# res4 <- topTable(fit, coef = "TreatmentGILT...DEC-TreatmentGILT", number = Inf) %>%
#   mutate(featureID = rownames(.)) %>%
#   arrange(P.Value)

############################ Saving tables for further use ##################################
# write.table(phos.diffex.pre, "Ex17 Phospho Differential Expression Pre-Treatment - Healthy.txt",
#             sep = "\t", row.names = F)
# 
# write.table(phos.diffex.gilt, "Ex17 Phospho Differential Expression Gilteritinib - Healthy.txt",
#             sep = "\t", row.names = F)
# 
# write.table(phos.diffex.gilt.dec, "Ex17 Phospho Differential Expression Gilteritinib Decitabine - Healthy.txt",
#             sep = "\t", row.names = F)
# 
# write.table(phos.paired.gilt.vs.pre, "Ex17 Phospho Differential Expression Gilteritinib - Pre-Treatment (paired).txt",
#             sep = "\t", row.names = F)
# 
# write.table(phos.paired.giltdec.vs.gilt, "Ex17 Phospho Differential Expression Gilteritinib Decitabine - Gilteritinib (paired).txt",
#             sep = "\t", row.names = F)

```


```{r computing KSEA, eval=FALSE, include=FALSE}
## Run this to make the pictures. We use PhosphoSite Plus + NetworKIN

set.seed(117)
ksea.pre <- phos.diffex.pre %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Pre-Treatment", ksea_FDR = 0.05)

set.seed(117)
ksea.gilt <- phos.diffex.gilt %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Gilterinib", ksea_FDR = 0.05)

set.seed(117)
ksea.giltdec <- phos.diffex.gilt.dec %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Gilterinib & Decitabine", ksea_FDR = 0.05)

set.seed(117)
ksea.gilt.vs.pre <- phos.paired.gilt.vs.pre %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Pre-Treatment-vs-Gilteritinib", ksea_FDR = 0.05)

set.seed(117)
ksea.giltdec.vs.gilt <- phos.paired.giltdec.vs.gilt %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Gilteritinib-vs-Gilteritinib & Decitabine", ksea_FDR = 0.05)

```

Below we collect the most significant differentially expressed genes using the comparisons stated above, 
as well as a KSEA plot made using PhosphoSite Plus + NetworKIN.
With a logarithmic scale we collect the adjusted p-values for each Kinase. 
Note the KSEA plot shows only those Kinases with at least 5 substrates found in the data, 
in addition to having an adjusted p-value below 0.05.

### Healthy vs Pre-Treatment

We compare the healthy donor samples to the Pre-Treatment samples.
Red indicates increased activity in Pre-Treatment samples, while blue shows increased activity
in healthy samples.

```{r Healthy vs Pre-Treatment}
show <- phos.diffex.pre %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Pre-Treatment-ksea-plot.png")
```

### Healthy vs Gilterinib

Next, we compare healthy samples to Gilteritinib treated samples.
To avoid over-representing any patient, we choose the earliest sample per patient for the Gilteritinib group.
Red indicates increased activity in Gilteritinib samples, while blue shows increased activity
in healthy samples.

```{r Healthy vs Gilterinib}
show <- phos.diffex.gilt %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Gilterinib-ksea-plot.png")
```

### Healthy vs Gilteritinib + Decitabine

Next, we compare healthy samples to Gilteritinib + Decitabine treated samples.
To avoid over-representing any patient, we choose the earliest sample per patient for the Gilteritinib +Decitabine group.
Red indicates increased activity in Gilteritinib + Decitabine samples, while blue shows increased activity
in healthy samples.

```{r Healthy vs Gilteritinib + Decitabine}
show <- phos.diffex.gilt.dec %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Gilterinib & Decitabine-ksea-plot.png")
```

### Pre-Treatment vs Gilteritinib

Next, we compare Pre-Treatment samples to Gilteritinib. 
To avoid over-representing any patient, we choose the earliest sample per patient for the Gilteritinib group.
The differential expression table is the result of paired sample t-tests.
Red indicates increased activity in Gilteritinib samples, while blue shows increased activity
in Pre-Treatment samples.

```{r Pre-Treatment vs Gilteritinib}
show <- phos.paired.gilt.vs.pre %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedPre-Treatment-vs-Gilteritinib-ksea-plot.png")
```

### Gilteritinib vs Gilteritinib + Decitabine

Finally, we compare Gilteritinib samples to Gilteritinib + Decitabine samples. 
To avoid over-representing any patient, we choose the earliest sample per patient for both the groups.
The Gilteritinib group in this comparison comprises only patients with a corresponding Gilteritinib + Decitabine sample,
and differential expression table is the result of paired sample t-tests.
Red indicates increased activity in Gilteritinib + Decitabine samples, while blue shows increased activity
in Gilteritinib samples.

```{r Gilteritinib vs Gilteritinib + Decitabine}
show <- phos.paired.giltdec.vs.gilt %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedGilteritinib-vs-Gilteritinib & Decitabine-ksea-plot.png")
```
