---
title: "Sunil Ex17 Phospho Correlation Enrichment"
author: "Camilo Posso"
date: "08/17/2021"
output: 
  html_document:
    code_folding: hide
    toc: true
---

[comment]: # (The KSEA plots are the same as those seen in the Differential Expression and KSEA report)

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, include=FALSE}
library(amlresistancenetworks)
library(leapr)
library(kableExtra)
library(dplyr)
library(readr)
library(readxl)
source("../../Util/synapseUtil.R")

# corrected for loading mass and plexID, long time and subtle
p.site.corrected <- querySynapseTable("syn26047149")

phos.dat <- p.site.corrected
phos.dat$Peptide <- sub("^.*@(.*)$", "\\1", phos.dat$Accession)
phos.dat$site <- sub("^(.*)@.*$", "\\1", phos.dat$Accession)
phos.dat$Gene <- sub("(.*)-.*$", "\\1", phos.dat$site)

## We have the metadata selected from the long form table
summary <- p.site.corrected[p.site.corrected$Accession == "AAK1-T606t@K.VQTT*PPPAVQGQK.V", 
                            c("Sample", "BeatAML Patient ID", 
                              "Plex", "Loading Mass", "Description", 
                              "Period", "Treatment")]

rownames(summary) <- summary$Sample

## Organizing time points into the right order (not alphanumeric)
time.points <- c("P2 Pre-Study", "P2 Cycle 1 Day 1", "P2 Cycle 1 Day 28", "P2 Cycle 2 Day 28", "P2 Cycle 3 Day 28", 
                 "P1B Cycle 1 Day 1", "P1B Cycle 1 Day 28", "P1B Cycle 2 Day 28", "P1B Cycle 3 Day 28", "P1B Cycle 6 Day 28",
                 "P1B Cycle 9 Day 28", "P1B Conc Cycle 9 Day 28") %>%
  factor(.)


summary$Description <- factor(summary$Description, levels = time.points)
summary <- summary[sort(summary$Sample), ]
summary$`BeatAML Patient ID` <- factor(summary$`BeatAML Patient ID`, levels = unique(summary$`BeatAML Patient ID`))
summary$Treatment <- make.names(summary$Treatment)
```


```{r eval=FALSE, include=FALSE}
KSDB <- read.csv(system.file("PSP&NetworKIN_Kinase_Substrate_Dataset_July2016.csv", 
                             package = "amlresistancenetworks"), stringsAsFactors = FALSE)

## Modifying the format of KSDB in order to run Correlation enrichment
kinase.members <- KSDB %>%
  select(KINASE, SUB_GENE, SUB_MOD_RSD) %>%
  mutate(Site = paste(SUB_GENE, SUB_MOD_RSD, sep = "-")) %>%
  group_by(KINASE) %>%
  summarise(Members = paste(Site, sep = "; ", collapse = "; "),
            sizes = length(Site)) %>%
  mutate(loose = strsplit(Members, "; "))

Max <- max(kinase.members$sizes) + 1
padded <- lapply(kinase.members$loose, 
                 FUN = function(x) append(x, rep("null", Max-length(x))))

names <- kinase.members$KINASE
desc <- names
sizes <- kinase.members$sizes
names(sizes) <- names
matrix <- matrix(unlist(padded), ncol = Max, nrow = length(names), byrow = TRUE)

PhosphositeNetworKIN <- list(names = names, 
                             desc = rep("null", length(names)), 
                             sizes = sizes, 
                             matrix = matrix)
```


### Get dataset

We access synapse to load the data in long form. We use data which has been corrected for both
Loading mass effect, as well as plex effect. Below we see a metadata table for this experiment.

```{r pressure, echo=FALSE}
show <- summary
kbl(show) %>%
  kable_paper("hover", full_width = T)
```

Here we perform 4 comparisons. We compare the healthy group to the treated, 
treated & early, and treated & late groups. We also compare the treated & early vs 
treated & late groups.

```{r prepare ratios and limma, eval=FALSE, include=FALSE}
##select the samples of interest
healthy <- subset(summary, `BeatAML Patient ID` == 'Healthy Donor Stroma')

pre.treatment <- subset(summary, Period == "Pre-Treatment")

## Earliest sample, per patient, which has been treated with GILT
treated.gilt <- subset(summary, Treatment == "GILT") %>%
  filter(`BeatAML Patient ID` != "5029") %>%
  group_by(`BeatAML Patient ID`) %>%
  slice(1)

## Earliest sample, per patient, which has been treated with GILT + DEC
treated.gilt.and.dec <- subset(summary, Treatment == "GILT...DEC") %>%
  group_by(`BeatAML Patient ID`) %>%
  slice(1)

## Gilteritinib samples with Gilt + Dec counterpart in same patient
treated.gilt.restricted <- subset(summary, Treatment == "GILT") %>%
  filter(`BeatAML Patient ID` %in% treated.gilt.and.dec$`BeatAML Patient ID`)

##then we spread the proteomics data into a matrix
phos.mat <- phos.dat%>%
  select(LogRatio, Sample, site)%>%
  tidyr::pivot_wider(values_from='LogRatio',names_from='Sample',
                      values_fn=list(LogRatio=mean))%>%
  tibble::column_to_rownames('site')

gene.to.site<-dplyr::select(phos.dat,Gene,site,Peptide)%>%distinct()%>%
    dplyr::mutate(residue=stringr::str_replace(site,paste0(Gene,'-'),''))%>%
    dplyr::mutate(residue=stringr::str_replace_all(residue,"([STY])", ";\\1"))%>%
    dplyr::mutate(residue=stringr::str_replace(residue,"^;", ""))%>%
    dplyr::mutate(residue=stringr::str_replace_all(residue,"([sty])", ""))

residues <- gene.to.site$residue
split.residues <- strsplit(residues, split = ";")
lengths <- sapply(split.residues, length)
single.residues <- data.frame(site = rep(gene.to.site$site, lengths), 
                              Gene = rep(gene.to.site$Gene, lengths), 
                              Residue = unlist(split.residues)) %>%
  mutate(`single site` = paste(Gene, Residue, sep = "-"))

# Split each phosphosite, ie GENE-S123S125 is split into GENE-S123, GENE-S125
phos.dat.single.site <- merge(phos.dat, single.residues)

phos.mat <- phos.dat.single.site %>%
  select(LogRatio, Sample, `single site`, site) %>%
  tidyr::pivot_wider(values_from='LogRatio',names_from='Sample',
                      values_fn=list(LogRatio=mean)) %>%
  mutate(count = rep(1,nrow(.))) 

single.site <- phos.mat$`single site`

# In correlation enrichment, these duplicates could artificially inflate a Kinase correlation
# One option is to average the correlations on the duplicates, however averaging the duplicates
# Prior to correlation enrichment is roughly the same, and much easier.

# Average duplicate phosphosites (duplicates arise from splitting compound sites, eg GENE-S123S125).
# Function rowsum is adding column-wise according to grouping in single.site.
phos.mat <- phos.mat %>%
  select(everything(),-`single site`, -site) %>%
  rowsum(single.site) %>%
  sweep(1, .$count, FUN = "/")
```

In this short report we collect both KSEA for each stated comparison, as well the results of
Jason McDermott's correlation enrichment. 

Correlation enrichment aims to identify pathways whose *genes* are highly correlated.
Within a group of samples, this approach can help identify pathways whose genes shows a high degree of concordance.
For a pathway \(\mathcal{P}\), we compute the correlation between all 
*pairs of genes* belonging to \(\mathcal{P}\), and define the `Correlation Enrichment` 
as the average correlation over all pairs. Values closer to 1 indicate that the genes 
in \(\mathcal{P}\) are tightly correlated; this is the value we plot for each pathway below. 
The p-value measures whether the difference in between the `Correlation Enrichment` and the 
background average correlation among genes is significant.

$$\\[.2in]$$


### Healthy vs Pre-Treatment

#### KSEA

Below we have a KSEA plot computed using the Phosphosite Plus + NetworKIN pathway
database. We're comparing the healthy donor samples to Pre-Treatment samples. 
Red indicates increased activity in Pre-Treatment samples, while blue shows increased activity
in healthy samples. 

```{r, include=TRUE, fig.align="center", fig.cap=c("Healthy vs Pre-Treatment")}
knitr::include_graphics("sig-includedHealthy-vs-Pre-Treatment-ksea-plot.png")
```

#### Correlation Enrichment

Below we show the correlation enrichment in Healthy samples on the left, 
as well as Pre-Treatment samples on the right. Pathways with higher enrichment score indicate more 
concordance among the genes in that pathway. Note that color does not have any
meaning in correlation enrichment.


```{r, echo=FALSE,out.width="49%", out.height = "49%", fig.cap="Left: Healthy - Right: Pre-Treatment",fig.show='hold',fig.align='center'}
knitr::include_graphics(c("sig-included-Healthy-correlation-enrichment-plot.png",
                          "sig-included-Pre-Treatment-correlation-enrichment-plot.png"))
``` 

$$\\[0.2in]$$

### Healthy vs Gilteritinib

#### KSEA

Below we have a KSEA plot computed using the Phosphosite Plus + NetworKIN pathway
database. We compare healthy samples to Gilteritinib treated samples.
To avoid over-representing any patient, we choose the earliest sample per patient for the Gilteritinib group.
Red indicates increased activity in Gilteritinib samples, while blue shows increased activity
in healthy samples.

```{r, include=TRUE, fig.align="center", fig.cap=c("Healthy vs Gilteritinib")}
knitr::include_graphics("sig-includedHealthy-vs-Gilterinib-ksea-plot.png")
```

#### Correlation Enrichment

Below we show the correlation enrichment in Healthy samples on the left, 
as well as Gilteritinib samples on the right. Pathways with higher enrichment score indicate more 
concordance among the genes in that pathway. Note that color does not have any
meaning in correlation enrichment.

```{r, echo=FALSE,out.width="49%", out.height = "49%", fig.cap="Left: Healthy - Right: Gilteritinib",fig.show='hold',fig.align='center'}
knitr::include_graphics(c("sig-included-Healthy-correlation-enrichment-plot.png",
                          "sig-included-Gilteritinib-correlation-enrichment-plot.png"))
``` 

$$\\[0.2in]$$

### Healthy vs Gilteritinib + Decitabine

#### KSEA

Below we have a KSEA plot computed using the Phosphosite Plus + NetworKIN pathway
database. Next, we compare healthy samples to Gilteritinib + Decitabine treated samples.
To avoid over-representing any patient, we choose the earliest sample per patient for the Gilteritinib Decitabine group.
Red indicates increased activity in Gilteritinib Decitabine samples, while blue shows increased activity
in healthy samples.

```{r, include=TRUE, fig.align="center", fig.cap=c("Healthy vs Gilteritinib Decitabine")}
knitr::include_graphics("sig-includedHealthy-vs-Gilterinib & Decitabine-ksea-plot.png")
```

#### Correlation Enrichment

Below we show the correlation enrichment in Healthy samples on the left, 
as well as Gilteritinib + Decitabine samples on the right. Pathways with higher enrichment score indicate more 
concordance among the genes in that pathway. Note that color does not have any
meaning in correlation enrichment.


```{r, echo=FALSE,out.width="49%", out.height = "49%", fig.cap="Left: Healthy - Right: Gilteritinib Decitabine",fig.show='hold',fig.align='center'}
knitr::include_graphics(c("sig-included-Healthy-correlation-enrichment-plot.png",
                          "sig-included-Gilteritinib Decitabine-correlation-enrichment-plot.png"))
``` 

$$\\[0.2in]$$

### Pre-Treatment vs Gilteritinib

#### KSEA

Below we have a KSEA plot computed using the Phosphosite Plus + NetworKIN pathway
database. Next, we compare Pre-Treatment samples to Gilteritinib treated samples.
To avoid over-representing any patient, we choose the earliest sample per patient for the Gilteritinib group.
Red indicates increased activity in Gilteritinib samples, while blue shows increased activity
in Pre-Treatment.

```{r, include=TRUE, fig.align="center", fig.cap=c("Pre-Treatment vs Gilteritinib")}
knitr::include_graphics("sig-includedPre-Treatment-vs-Gilteritinib-ksea-plot.png")
```

#### Correlation Enrichment

Below we show the correlation enrichment in Pre-Treatment samples on the left, 
as well as Gilteritinib samples on the right. Pathways with higher enrichment score indicate more 
concordance among the genes in that pathway. Note that color does not have any
meaning in correlation enrichment.

```{r, echo=FALSE,out.width="49%", out.height = "49%", fig.cap="Left: Pre-Treatment - Right: Gilteritinib",fig.show='hold',fig.align='center'}
knitr::include_graphics(c("sig-included-Pre-Treatment-correlation-enrichment-plot.png",
                          "sig-included-Gilteritinib-correlation-enrichment-plot.png"))
``` 

$$\\[.2in]$$

### Gilteritinib vs Gilteritinib + Decitabine

#### KSEA

Below we have a KSEA plot computed using the Phosphosite Plus + NetworKIN pathway
database. Next, we compare Gilteritinib samples to Gilteritinib + Decitabine treated samples.
To avoid over-representing any patient, we choose the earliest sample per patient in both groups.
The Gilteritinib group in this comparison comprises only patients with a corresponding Gilteritinib + Decitabine sample.
Red indicates increased activity in Gilteritinib + Decitabine samples, while blue shows increased activity
in Gilteritinib samples.

```{r, include=TRUE, fig.align="center", fig.cap=c("Gilteritinib vs Gilteritinib + Decitabine")}
knitr::include_graphics("sig-includedGilteritinib-vs-Gilteritinib & Decitabine-ksea-plot.png")
```

#### Correlation Enrichment

Below we show the correlation enrichment in Gilteritinib samples on the left, 
as well as Gilteritinib + Decitabine samples on the right. Pathways with higher enrichment score indicate more 
concordance among the genes in that pathway. Note that color does not have any
meaning in correlation enrichment.

```{r, echo=FALSE,out.width="49%", out.height = "49%", fig.cap="Left: Gilteritinib - Right: Gilteritinib + Decitabine",fig.show='hold',fig.align='center'}
knitr::include_graphics(c("sig-included-Gilteritinib Restricted-correlation-enrichment-plot.png",
                          "sig-included-Gilteritinib Decitabine-correlation-enrichment-plot.png"))
``` 

$$\\[.2in]$$


[comment]: # (Below is the code to produce the enrichment correlation plots seen in this markdown)

```{r enrichment correlation, eval=FALSE, include=FALSE}
set.seed(117)
enrichment.correlation.healthy <- plotCorrelationEnrichment(exprs = phos.mat[, healthy$Sample], 
                                                            prefix = "Healthy", pathway.plot.size = 4, 
                                                            geneset = kinasesubstrates, corr.cutoff = 0.4,
                                                            clean.names = FALSE, height = 12)
set.seed(117)
enrichment.correlation.pre <- plotCorrelationEnrichment(exprs = phos.mat[,pre.treatment$Sample], 
                                                        prefix = "Pre-Treatment", pathway.plot.size = 4,
                                                        geneset = kinasesubstrates, corr.cutoff = 0.4,
                                                        clean.names = FALSE, height = 12)
set.seed(117)
enrichment.correlation.treated.gilt <- plotCorrelationEnrichment(exprs = phos.mat[,treated.gilt$Sample], 
                                                                 prefix = "Gilteritinib", pathway.plot.size = 4,
                                                                 geneset = kinasesubstrates, corr.cutoff = 0.4,
                                                                 clean.names = FALSE, height = 12)
set.seed(117)
enrichment.correlation.treated.giltdec <- plotCorrelationEnrichment(exprs = phos.mat[, treated.gilt.and.dec$Sample], 
                                                                    prefix = "Gilteritinib Decitabine", pathway.plot.size = 4,
                                                                    geneset = kinasesubstrates, corr.cutoff = 0.4,
                                                                    clean.names = FALSE, height = 12)
set.seed(117)
enrichment.correlation.treated.gilt.restricted <- plotCorrelationEnrichment(exprs = phos.mat[, treated.gilt.restricted$Sample], 
                                                                            prefix = "Gilteritinib Restricted", 
                                                                            pathway.plot.size = 4,
                                                                            geneset = kinasesubstrates, corr.cutoff = 0.4,
                                                                            clean.names = FALSE, height = 12)
```
